### ISSUE TITLE
7.0.1 build 3 causes error on import

### ISSUE THREAD

**Issue by robert-schmidtke** [conda-forge/arrow-cpp-feedstock#865]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Build 3 over build 2 makes the following changes to google libraries (inspecting the `meta.yaml` files)
```diff
97c97
<     - google-cloud-cpp 2.1.0 h838d150_3
---
>     - google-cloud-cpp 2.2.0 h838d150_0
```

Which causes in `ImportError`
```python
import pyarrow as pa
  File "/home/rschmidtke/miniconda3/conda-bld/tsawsdatahub_1663327389681/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/pyarrow/__init__.py", line 65, in <module>
    import pyarrow.lib as _lib
ImportError: /home/rschmidtke/miniconda3/conda-bld/tsawsdatahub_1663327389681/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/lib/python3.10/site-packages/pyarrow/../../../libarrow.so.700: undefined symbol: _ZN6google5cloud6v2_2_06StatusD1Ev
```

Manually pinning `google-cloud-cpp` to `2.1.0` fixes the issue. It seems it may be necessary to restrict the library here as well?

### Installed packages

```shell
_libgcc_mutex:            0.1-conda_forge             conda-forge
    _openmp_mutex:            4.5-2_gnu                   conda-forge
    aenum:                    2.2.6-pyhd3deb0d_0          conda-forge
    aiohttp:                  3.8.1-py310h5764c6d_1       conda-forge
    aiosignal:                1.2.0-pyhd8ed1ab_0          conda-forge
    appdirs:                  1.4.4-pyh9f0ad1d_0          conda-forge
    arrow-cpp:                7.0.1-py310h028ac91_3_cpu   conda-forge
    asn1crypto:               1.5.1-pyhd8ed1ab_0          conda-forge
    async-timeout:            4.0.2-pyhd8ed1ab_0          conda-forge
    attrs:                    22.1.0-pyh71513ae_1         conda-forge
    aws-c-cal:                0.5.11-h95a6274_0           conda-forge
    aws-c-common:             0.6.2-h7f98852_0            conda-forge
    aws-c-event-stream:       0.2.7-h3541f99_13           conda-forge
    aws-c-io:                 0.10.5-hfb6a706_0           conda-forge
    aws-checksums:            0.1.11-ha31a3da_7           conda-forge
    aws-sam-translator:       1.51.0-pyhd8ed1ab_0         conda-forge
    aws-sdk-cpp:              1.8.186-hb4091e7_3          conda-forge
    aws-xray-sdk:             2.10.0-pyhd8ed1ab_0         conda-forge
    awswrangler:              2.16.1-pyhd8ed1ab_0         conda-forge
    backoff:                  1.11.1-pyhd8ed1ab_0         conda-forge
    beautifulsoup4:           4.11.1-pyha770c72_0         conda-forge
    boto3:                    1.24.74-pyhd8ed1ab_0        conda-forge
    botocore:                 1.27.74-pyhd8ed1ab_0        conda-forge
    brotlipy:                 0.7.0-py310h5764c6d_1004    conda-forge
    bzip2:                    1.0.8-h7f98852_4            conda-forge
    c-ares:                   1.18.1-h7f98852_0           conda-forge
    ca-certificates:          2022.9.14-ha878542_0        conda-forge
    certifi:                  2022.9.14-pyhd8ed1ab_0      conda-forge
    cffi:                     1.15.1-py310h255011f_0      conda-forge
    cfn-lint:                 0.65.0-pyhd8ed1ab_0         conda-forge
    charset-normalizer:       2.0.12-pyhd8ed1ab_0         conda-forge
    click:                    8.1.3-py310hff52083_0       conda-forge
    cryptography:             37.0.4-py310h597c629_0      conda-forge
    decorator:                5.1.1-pyhd8ed1ab_0          conda-forge
    docker-py:                5.0.3-py310hff52083_2       conda-forge
    docker-pycreds:           0.4.0-py_0                  conda-forge
    ecdsa:                    0.18.0-pyhd8ed1ab_1         conda-forge
    et_xmlfile:               1.0.1-py_1001               conda-forge
    flask:                    2.1.3-pyhd8ed1ab_0          conda-forge
    flask_cors:               3.0.10-pyhd3deb0d_0         conda-forge
    frozenlist:               1.3.1-py310h5764c6d_0       conda-forge
    gflags:                   2.2.2-he1b5a44_1004         conda-forge
    glog:                     0.6.0-h6f12383_0            conda-forge
    gmp:                      6.2.1-h58526e2_0            conda-forge
    gmpy2:                    2.1.2-py310h92f7908_0       conda-forge
    graphql-core:             3.2.1-pyhd8ed1ab_0          conda-forge
    gremlinpython:            3.5.2-pyhd8ed1ab_0          conda-forge
    grpc-cpp:                 1.47.1-h05bd8bd_6           conda-forge
    icu:                      70.1-h27087fc_0             conda-forge
    idna:                     3.3-pyhd8ed1ab_0            conda-forge
    importlib-metadata:       4.11.4-py310hff52083_0      conda-forge
    importlib_metadata:       4.11.4-hd8ed1ab_0           conda-forge
    importlib_resources:      3.3.1-pyhd8ed1ab_1          conda-forge
    iniconfig:                1.1.1-pyh9f0ad1d_0          conda-forge
    isodate:                  0.6.1-pyhd8ed1ab_0          conda-forge
    itsdangerous:             2.1.2-pyhd8ed1ab_0          conda-forge
    jinja2:                   3.1.2-pyhd8ed1ab_1          conda-forge
    jmespath:                 1.0.1-pyhd8ed1ab_0          conda-forge
    jschema-to-python:        1.2.3-pyhd8ed1ab_0          conda-forge
    jsondiff:                 2.0.0-pyhd8ed1ab_0          conda-forge
    jsonpatch:                1.32-pyhd8ed1ab_0           conda-forge
    jsonpath-ng:              1.5.3-pyhd8ed1ab_0          conda-forge
    jsonpickle:               2.2.0-pyhd8ed1ab_0          conda-forge
    jsonpointer:              2.0-py_0                    conda-forge
    jsonschema:               3.2.0-pyhd8ed1ab_3          conda-forge
    junit-xml:                1.9-pyh9f0ad1d_0            conda-forge
    keyutils:                 1.6.1-h166bdaf_0            conda-forge
    krb5:                     1.19.3-h3790be6_0           conda-forge
    ld_impl_linux-64:         2.36.1-hea4e1c9_2           conda-forge
    libabseil:                20220623.0-cxx17_h48a1fff_4 conda-forge
    libblas:                  3.9.0-16_linux64_openblas   conda-forge
    libbrotlicommon:          1.0.9-h166bdaf_7            conda-forge
    libbrotlidec:             1.0.9-h166bdaf_7            conda-forge
    libbrotlienc:             1.0.9-h166bdaf_7            conda-forge
    libcblas:                 3.9.0-16_linux64_openblas   conda-forge
    libcrc32c:                1.1.2-h9c3ff4c_0            conda-forge
    libcurl:                  7.83.1-h7bff187_0           conda-forge
    libedit:                  3.1.20191231-he28a2e2_2     conda-forge
    libev:                    4.33-h516909a_1             conda-forge
    libevent:                 2.1.10-h9b69904_4           conda-forge
    libffi:                   3.4.2-h7f98852_5            conda-forge
    libgcc-ng:                12.1.0-h8d9b700_16          conda-forge
    libgfortran-ng:           12.1.0-h69a702a_16          conda-forge
    libgfortran5:             12.1.0-hdcd56e2_16          conda-forge
    libgomp:                  12.1.0-h8d9b700_16          conda-forge
    libgoogle-cloud:          2.2.1-h838d150_0            conda-forge
    libiconv:                 1.16-h516909a_0             conda-forge
    liblapack:                3.9.0-16_linux64_openblas   conda-forge
    libnghttp2:               1.47.0-hdcd2b5c_1           conda-forge
    libnsl:                   2.0.0-h7f98852_0            conda-forge
    libopenblas:              0.3.21-pthreads_h78a6416_3  conda-forge
    libprotobuf:              3.21.6-h6239696_0           conda-forge
    libsqlite:                3.39.3-h753d276_0           conda-forge
    libssh2:                  1.10.0-haa6b8db_3           conda-forge
    libstdcxx-ng:             12.1.0-ha89aaad_16          conda-forge
    libthrift:                0.16.0-h491838f_2           conda-forge
    libutf8proc:              2.7.0-h7f98852_0            conda-forge
    libuuid:                  2.32.1-h7f98852_1000        conda-forge
    libxml2:                  2.9.14-h22db469_4           conda-forge
    libxslt:                  1.1.35-h8affb1d_0           conda-forge
    libzlib:                  1.2.12-h166bdaf_3           conda-forge
    lxml:                     4.9.1-py310h5764c6d_0       conda-forge
    lz4-c:                    1.9.3-h9c3ff4c_1            conda-forge
    markupsafe:               2.1.1-py310h5764c6d_1       conda-forge
    moto:                     4.0.3-pyhd8ed1ab_0          conda-forge
    mpc:                      1.2.1-h9f54685_0            conda-forge
    mpfr:                     4.1.0-h9202a9a_1            conda-forge
    multidict:                6.0.2-py310h5764c6d_1       conda-forge
    ncurses:                  6.3-h27087fc_1              conda-forge
    nest-asyncio:             1.5.5-pyhd8ed1ab_0          conda-forge
    networkx:                 2.8.6-pyhd8ed1ab_0          conda-forge
    numpy:                    1.23.3-py310h53a5b5f_0      conda-forge
    openapi-schema-validator: 0.2.3-pyhd8ed1ab_0          conda-forge
    openapi-spec-validator:   0.4.0-pyhd8ed1ab_1          conda-forge
    openpyxl:                 3.0.10-py310h5764c6d_1      conda-forge
    opensearch-py:            1.1.0-pyhd8ed1ab_0          conda-forge
    openssl:                  1.1.1q-h166bdaf_0           conda-forge
    orc:                      1.7.6-h09e0d61_1            conda-forge
    packaging:                21.3-pyhd8ed1ab_0           conda-forge
    pandas:                   1.4.4-py310h769672d_0       conda-forge
    parquet-cpp:              1.5.1-2                     conda-forge
    pbr:                      5.10.0-pyhd8ed1ab_0         conda-forge
    pg8000:                   1.29.1-pyhd8ed1ab_0         conda-forge
    pip:                      22.2.2-pyhd8ed1ab_0         conda-forge
    pluggy:                   1.0.0-py310hff52083_3       conda-forge
    ply:                      3.11-py_1                   conda-forge
    progressbar2:             4.0.0-pyhd8ed1ab_0          conda-forge
    py:                       1.11.0-pyh6c4a22f_0         conda-forge
    pyarrow:                  7.0.1-py310h225c066_3_cpu   conda-forge
    pyasn1:                   0.4.8-py_0                  conda-forge
    pycparser:                2.21-pyhd8ed1ab_0           conda-forge
    pymysql:                  1.0.2-pyhd8ed1ab_0          conda-forge
    pyodbc:                   4.0.34-py310hd8f1fbe_0      conda-forge
    pyopenssl:                22.0.0-pyhd8ed1ab_0         conda-forge
    pyparsing:                3.0.9-pyhd8ed1ab_0          conda-forge
    pyrsistent:               0.18.1-py310h5764c6d_1      conda-forge
    pysocks:                  1.7.1-pyha2e5f31_6          conda-forge
    pytest:                   7.1.3-py310hff52083_0       conda-forge
    python:                   3.10.6-h582c2e5_0_cpython   conda-forge
    python-dateutil:          2.8.2-pyhd8ed1ab_0          conda-forge
    python-jose:              3.3.0-pyh6c4a22f_1          conda-forge
    python-utils:             3.3.3-pyhd8ed1ab_0          conda-forge
    python_abi:               3.10-2_cp310                conda-forge
    pytz:                     2022.1-pyhd8ed1ab_0         conda-forge
    pyyaml:                   6.0-py310h5764c6d_4         conda-forge
    re2:                      2022.06.01-h27087fc_0       conda-forge
    readline:                 8.1.2-h0f457ee_0            conda-forge
    redshift_connector:       2.0.908-pyhd8ed1ab_0        conda-forge
    requests:                 2.28.0-pyhd8ed1ab_1         conda-forge
    requests-aws4auth:        1.1.1-pyhd8ed1ab_0          conda-forge
    responses:                0.21.0-pyhd8ed1ab_0         conda-forge
    rsa:                      4.9-pyhd8ed1ab_0            conda-forge
    s2n:                      1.0.10-h9b69904_0           conda-forge
    s3transfer:               0.6.0-pyhd8ed1ab_0          conda-forge
    sarif-om:                 1.0.4-pyhd8ed1ab_0          conda-forge
    scramp:                   1.4.1-pyhd8ed1ab_0          conda-forge
    setuptools:               65.3.0-pyhd8ed1ab_1         conda-forge
    six:                      1.16.0-pyh6c4a22f_0         conda-forge
    snappy:                   1.1.9-hbd366e4_1            conda-forge
    soupsieve:                2.3.2.post1-pyhd8ed1ab_0    conda-forge
    sshpubkeys:               3.3.1-pyhd8ed1ab_0          conda-forge
    tk:                       8.6.12-h27826a3_0           conda-forge
    tomli:                    2.0.1-pyhd8ed1ab_0          conda-forge
    tsawsdatahub:             0.0.1-py_1                  local
    typing-extensions:        4.3.0-hd8ed1ab_0            conda-forge
    typing_extensions:        4.3.0-pyha770c72_0          conda-forge
    tzdata:                   2022c-h191b570_0            conda-forge
    unixodbc:                 2.3.10-h583eb01_0           conda-forge
    urllib3:                  1.26.11-pyhd8ed1ab_0        conda-forge
    websocket-client:         1.4.1-pyhd8ed1ab_0          conda-forge
    werkzeug:                 2.1.2-pyhd8ed1ab_1          conda-forge
    wheel:                    0.37.1-pyhd8ed1ab_0         conda-forge
    wrapt:                    1.14.1-py310h5764c6d_0      conda-forge
    xmltodict:                0.13.0-pyhd8ed1ab_0         conda-forge
    xz:                       5.2.6-h166bdaf_0            conda-forge
    yaml:                     0.2.5-h7f98852_2            conda-forge
    yarl:                     1.7.2-py310h5764c6d_2       conda-forge
    zipp:                     3.8.1-pyhd8ed1ab_0          conda-forge
    zlib:                     1.2.12-h166bdaf_3           conda-forge
    zstd:                     1.5.2-h6239696_4            conda-forge
```

### Environment info

```shell
active environment : tsawsdatahub
    active env location : /home/rschmidtke/miniconda3/envs/tsawsdatahub
            shell level : 1
       user config file : /home/rschmidtke/.condarc
 populated config files : /home/rschmidtke/.condarc
          conda version : 4.14.0
    conda-build version : 3.22.0
         python version : 3.9.12.final.0
       virtual packages : __linux=5.10.102.1=0
                          __glibc=2.31=0
                          __unix=0=0
                          __archspec=1=x86_64
       base environment : /home/rschmidtke/miniconda3  (writable)
      conda av data dir : /home/rschmidtke/miniconda3/etc/conda
  conda av metadata url : None
           channel URLs : https://<redacted>.jfrog.io/artifactory/api/conda/conda-local/linux-64
                          https://<redacted>.jfrog.io/artifactory/api/conda/conda-local/noarch
                          https://conda.anaconda.org/conda-forge/linux-64
                          https://conda.anaconda.org/conda-forge/noarch
                          https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/rschmidtke/miniconda3/pkgs
                          /home/rschmidtke/.conda/pkgs
       envs directories : /home/rschmidtke/miniconda3/envs
                          /home/rschmidtke/.conda/envs
               platform : linux-64
             user-agent : conda/4.14.0 requests/2.28.1 CPython/3.9.12 Linux/5.10.102.1-microsoft-standard-WSL2 ubuntu/20.04.5 glibc/2.31 solver/libmamba conda-libmamba-solver/22.8.1 libmambapy/0.25.0
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False
```

**Comment by h-vetinari** [comment]
Thanks for the report. The issue here is `libgoogle-cloud:          2.2.1-h838d150_0` due to [this](https://github.com/conda-forge/google-cloud-cpp-feedstock/pull/116)

**Comment by h-vetinari** [comment]
I'll open a PR to mark 2.2.1 as broken as soon as I get to a computer in a couple hours, but anyone can do it and is welcome to (example [PR](https://github.com/conda-forge/admin-requests/pull/451) for reference).

**Comment by h-vetinari** [comment]
This will be fixed by https://github.com/conda-forge/admin-requests/pull/488. In the meantime you can add a constraint of <2.2.1 to get things working again.

**Comment by robert-schmidtke** [comment]
Thanks, that's super quick!

**Comment by h-vetinari** [comment]
@robert-schmidtke 
Could you check if this is still happening, and close the issue if not?

**Comment by robert-schmidtke** [comment]
Unfortunately I can only come back to this in two weeks time, but the
broken mark should fix it! I went with pinning the 2.1.x version instead.
I'll make a note and revisit and close the issue then. Many thanks, much
appreciated :)

**Comment by h-vetinari** [comment]
It seems the issue disappeared for the dask folks, so closing this. Please open a new issue if anything (still) goes wrong.
